lib/index.js

```diff
@@ -63,4 +105,130 @@ Playground.Piezo = {
   }
 };
 
+Playground.parseFirmataByte = function(data) {
+  // Parse a byte value from two 7-bit byte firmata response bytes.
+  if (data.length != 2) {
+    throw('Expected 2 bytes of firmata response for a byte value!');
+  }
+  return (data[0] & 0x7F) | ((data[1] & 0x01) << 7);
+};
+
+/**
+ * Parse a 4 byte floating point value from a 7-bit byte firmata response
+ * byte array.  Each pair of firmata 7-bit response bytes represents a single
+ * byte of float data so there should be 8 firmata response bytes total.
+ * @param {Array} data
+ */
+Playground.parseFirmataFloat = function(data) {
+  return parseFirmata(data, '<f');
+};
+
+Playground.parseFirmataLong = function(data) {
+  return parseFirmata(data, '<l');
+};
+
+function parseFirmata(data, format) {
+  if (data.length != 8) {
+    throw('Expected 8 bytes of firmata response for value to parse.');
+  }
+  //  Convert 2 7-bit bytes in little endian format to 1 8-bit byte for each
+  //    of the four floating point bytes.
+  var rawBytes = new ArrayBuffer(4);
+  for (var i = 0; i < 4; i++) {
+    rawBytes[i] = Playground.parseFirmataByte(data.slice(i * 2, i * 2 + 2));
+  }
+  return bufferpack.unpack(format, rawBytes)[0];
+}
+
+Playground.Accelerometer = {
+  initialize: {
+    value: function (opts, dataHandler) {
+      replyHandlers[CP_ACCEL_READ_REPLY] = function(data) {
+        if (data.length < 26) {
+          console.log('Received accelerometer response with not enough data!');
+          return
+        }
+        var x = Playground.parseFirmataFloat(data.slice(2, 10));
+        var y = Playground.parseFirmataFloat(data.slice(10, 18));
+        var z = Playground.parseFirmataFloat(data.slice(18, 26));
+        dataHandler({x: x, y: y, z: z});
```

Comment:
Once the [change here is made](https://github.com/rwaldron/playground-io/pull/2/files#r62227978), this becomes much simpler: 

```js
// data is already an array of 12 bytes: 4 bytes per axis
dataHandler({
  x: bufferpack.unpack('<f', data.slice(0, 4)), 
  y: bufferpack.unpack('<f', data.slice(4, 8)), 
  z: bufferpack.unpack('<f', data.slice(8, 12)), 
});
```


If you can use node.js Buffer, it's even easier...

```js
// data is already an array of 12 bytes: 4 bytes per axis
var buffer = new Buffer(data);

dataHandler({
  x: buffer.readFloatLE(0), 
  y: buffer.readFloatLE(4), 
  z: buffer.readFloatLE(8), 
});
```